### PR TITLE
Neural Network Design Doc

### PR DESCRIPTION
**Pull request by haying**
JIRA: MADLIB-424

And a minor doxygen fix of linear regression

### CODE REVIEW

File: doc/design/modules/neural-network.tex [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    &= \frac{\partial f}{\partial x_{k}^j} \cdot \phi'(\mathit{net}_{k}^j) \\
+    &= \sum_{t=1}^{n_{k+1}} \left( \frac{\partial f}{\partial \mathit{net}_{k+1}^t} \cdot \frac{\partial \mathit{net}_{k+1}^t}{\partial x_{k}^j} \right) \cdot \phi'(\mathit{net}_{k}^j) \\
+    &= \sum_{t=1}^{n_{k+1}} \left( \delta_{k+1}^t \cdot \frac{\partial \left( \sum_{s=0}^{n_{k}} x_{k}^s u_{k}^{st} \right) }{\partial x_{k}^j} \right) \cdot \phi'(\mathit{net}_{k}^j) \\
+    &= \sum_{t=1}^{n_{k+1}} \left( \delta_{k+1}^t \cdot u_{k}^{jt} \right) \cdot \phi'(\mathit{net}_{k}^j) \\
+\end{aligned}\]
+To sum up, we need the following equation for error back propagation
+\[\boxed{\delta_{k}^j = \sum_{t=1}^{n_{k+1}} \left( \delta_{k+1}^t \cdot u_{k}^{jt} \right) \cdot \phi'(\mathit{net}_{k}^j)}\]
+where $k = 1,...,N-1$, and $j = 1,...,n_{k}$.
+
+\subsubsection{The $\mathit{Gradient}$ Function}
```
Comment (haying) [inline doc/design/modules/neural-network.tex:92]: test: I am giving comment here to show examples for Grace.

**Comment by haying** [comment]
@haradh1 You can cherry-pick the fix of linregr here if we reject the design doc of neural net to go in v0.5

**Comment by haradh1** [comment]
@haying yep.

**Comment by haradh1** [comment]
Pulled  69630a6 only for now.

**Comment by cwelton** [comment]
This has been pulled into the outstanding work on neural nets.

This PR will likely not be pulled in directly, but rather through the change that eventually includes NN.  Leaving open until that work is resolved.

**Pull request closed without merging**